openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -125,18 +129,19 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
     if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
-        CourseHomeMessages.register_info_message(
-            request,
-            Text(_(
-                '{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
-            )).format(
-                open_enroll_link='',
-                close_enroll_link=''
-            ),
-            title=Text(_('Welcome to {course_display_name}')).format(
-                course_display_name=course.display_name
+        if not allow_anonymous:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    '{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
```

Comment:
Same question as above. If a course is marked for public access, how do students enrol?